src/main/java/eu/europeana/apikey/mail/MailService.java

```diff
@@ -20,28 +23,84 @@
     @Autowired
     public JavaMailSender emailSender;
 
-    private void sendSimpleMessage(String from, String[] bcc, String to, String subject, String messageBody) throws SendMailException {
+    private void sendSimpleMessage(SimpleMailMessage template, String messageBody) throws SendMailException {
         LOG.debug("Sending email ...");
         try {
             SimpleMailMessage message = new SimpleMailMessage();
-            message.setFrom(from);
-            message.setBcc(bcc);
-            message.setTo(to);
-            message.setSubject(subject);
+            message.setFrom(Objects.requireNonNull(template.getFrom()));
+            message.setBcc(Objects.requireNonNull(template.getBcc()));
+            message.setTo(Objects.requireNonNull(template.getTo())[0]);
+            message.setSubject(Objects.requireNonNull(template.getSubject()));
```

Comment:
Personally I feel it's more elegant to check if values or not null while loading the configuration and let start-up fail. That way you don't need to check it here, but this will work as well.